Modules/Wishlist/Http/Controllers/WishlistController.php

```diff
@@ -0,0 +1,502 @@
+                }
+                $board = Board::find($boardv->id);
+                if ($board) {
+                    $boardArr[$boardk]['board_key'] = $board->board_key;
+                    $boardArr[$boardk]['board_id'] = $board->id;
+                    $boardArr[$boardk]['board_name'] = $board->name;
+                    $boardArr[$boardk]['board_visibility'] = $board->visibility;
+                    $boardArr[$boardk]['board_image'] = $productImage;
+                    $boardArr[$boardk]['products_count'] = $wishesByBoardCount;
+                }
+            }
+        }
+
+        $data = $boardArr;
+
+        $response = ['res' => true, 'msg' => "", 'data' => $data];
+        return response()->json($response);
+    }
+
+    public function fetchBoard(Request $request, $key) {
+        $boardArr = [];
+        $productArr = [];
+        $board = Board::where('board_key', $key)->first();
+        if ($board) {
+            $wishesByBoard = Wishlist::where('board_id', $board->id)->where('cart_id', null)->get();
+            if ($wishesByBoard) {
+                foreach ($wishesByBoard as $wish) {
+                    $product = Products::find($wish->product_id);
+                    $productWholeSalePrice = $product->usd_wholesale_price;
+                    $productRetailPrice = $product->usd_retail_price;
+                    if ($wish->variant_id) {
+                        $variant = ProductVariation::find($wish->variant_id);
+                        $productWholeSalePrice = $variant->price;
+                        $productRetailPrice = $variant->retail_price;
+                    }
+                    $productDet['id'] = $wish->id;
+                    $productDet['product_id'] = $product->id;
+                    $productDet['product_name'] = $product->name;
+                    $productDet['product_wholesale_price'] = $productWholeSalePrice;
+                    $productDet['product_retail_price'] = $productRetailPrice;
+                    $productDet['product_qty'] = $wish->quantity;
+                    $productDet['style_name'] = $wish->style_name;
+                    $productDet['style_group_name'] = $wish->style_group_name;
+                    $productDet['type'] = $wish->type;
+                    $productDet['product_image'] = $product->featured_image != '' ? $product->featured_image : asset('public/img/logo-image.png');
+                    $productArr[] = $productDet;
+                }
+            }
+            $wishesByBrandCount = Wishlist::where('board_id', $board->id)->where('cart_id', null)->get()->count();
+            $board->products_count = $wishesByBrandCount;
+            $boardArr = $board;
+        }
+        $data = array(
+            'board_arr' => $boardArr,
+            'product_arr' => $productArr,
+        );
+        $response = ['res' => true, 'msg' => "", 'data' => $data];
+        return response()->json($response);
+    }
+
+    public function addBoard(Request $request) {
```

Comment:
phpdocs missing, and what is a board? Is it wishlish specific? Since you're add, updating and deleting boards so a better approach would be to create it as a separate module to make code more manageable.